client/app/routes/notifications_.$id.tsx

```diff
@@ -0,0 +1,25 @@
+import { LoaderFunctionArgs, json, redirect } from "@remix-run/node";
+import { checkAuth } from "~/lib/check-auth";
+import { prisma } from "~/lib/prisma.server";
+
+export const loader = async ({ request, params }: LoaderFunctionArgs) => {
+	const userId = await checkAuth(request);
+
+	const notificationId = Number(params.id as string);
+
+	const notification = await prisma.notification.findFirst({
+		where: { id: notificationId },
+	});
+
+	if (notification) {
```

Comment:
There's a pattern known as early return.

```suggestion
	if (!notification) {
	 throw new json({}, {status: 404})
	}
```

And then any code after will have `notification` not null